swap/src/bitcoin/wallet.rs

```diff
@@ -66,54 +111,56 @@ impl BuildTxLockPsbt for Wallet {
         output_address: Address,
         output_amount: Amount,
     ) -> Result<PartiallySignedTransaction> {
-        let psbt = self.inner.fund_psbt(output_address, output_amount).await?;
-        let as_hex = base64::decode(psbt)?;
-
-        let psbt = bitcoin::consensus::deserialize(&as_hex)?;
-
+        self.sync_wallet().await?;
+        tracing::debug!("building tx lock");
+        let (psbt, _details) = self.inner.lock().await.create_tx(
+            bdk::TxBuilder::with_recipients(vec![(
+                output_address.script_pubkey(),
+                output_amount.as_sat(),
+            )])
+            // todo: get actual fee
+            .fee_rate(FeeRate::from_sat_per_vb(5.0)),
```

Comment:
Move this value in a const so it's easier to remove all static fee values at once.